pika/adapters/libev_connection.py

```diff
@@ -237,12 +241,12 @@ def _timer_callback(self, timer, libev_events):
             (callback_method, callback_timeout,
              kwargs) = self._active_timers[timer]
 
+            self.remove_timeout(timer)
```

Comment:
Moved `remove_timeout` ahead of the callback to make sure it gets cleaned up, even if callback raises; eliminates need for `try/finally`